File: docusign-sdk-sample-objc/docusign-sdk-sample-objc/Managers/EnvelopesManager.m
//
//  EnvelopeManager.m
//  docusign-sdk-sample-objc
//
//  Copyright © 2017 DocuSign. All rights reserved.
//

#import "EnvelopesManager.h"

@interface EnvelopesManager ()

// properties
@property (nonatomic, retain) DSMEnvelopesManager *mDSMEnvelopesManager;
@property (nonatomic, retain) NSArray<DSMEnvelopeDefinition *> *mEnvelopeDefinitions;

@end


@implementation EnvelopesManager

+ (id) sharedInstance
{
    static EnvelopesManager * sharedManager = nil;
    
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        sharedManager = [[self alloc] init];

        if (sharedManager.mDSMEnvelopesManager == nil)
        {
            sharedManager.mDSMEnvelopesManager = [[DSMEnvelopesManager alloc] init];
        }

    });
    return sharedManager;
}


#pragma mark - Public Methods

// get list of cached envelope defitions
- (NSArray<DSMEnvelopeTemplateDefinition *> *) getCachedEnvelopesList
{
    return [self.mDSMEnvelopesManager cachedEnvelopes];
}


// get list of cached envelope ids
- (NSArray<NSString *> *) getCachedEnvelopeIds
{
    return [self.mDSMEnvelopesManager cachedEnvelopeIds];
}


// sync cached envelopes
- (void) syncCachedEnvelopes
{
    [self.mDSMEnvelopesManager syncEnvelopes];
}


// remove cached envelope by id
- (void) removeCachedEnvelopeById:(NSString *)envelopeId
{
    [self.mDSMEnvelopesManager removeCachedEnvelopeWithId:envelopeId];
}

// present compose envelope view controller
- (void) presentComposeEnvelopeViewControllerWithPresentingController:(UIViewController *)controller signingMode:(DSMSigningMode)signingMode {
    // you can use method hasCachedDraftEnvelope to determine whether you'd like to resume a draft or not
    [self.mDSMEnvelopesManager presentComposeEnvelopeControllerWithPresentingController:controller signingMode:signingMode resumeWithDraft:YES animated:YES completion:nil];
}

@end

File: docusign-sdk-sample-swift/docusign-sdk-sample-swift/Managers/EnvelopesManager.swift
//
//  EnvelopesManager.swift
//  docusign-sdk-sample-swift
//
//  
//  Copyright © 2017 DocuSign. All rights reserved.
//

import DocuSignSDK
import Foundation


class EnvelopesManager
{
    // singleton instance
    static let sharedInstance = EnvelopesManager()

    // DSM Envelopes Manager
    var mDSMEnvelopesManager: DSMEnvelopesManager?;
    
    // list of template definitions
    var mEnvelopeDefinitions: [DSMEnvelopeDefinition]?;

    //This prevents others from using the default '()' initializer for this class.
    private init()
    {
        if (self.mDSMEnvelopesManager == nil)
        {
            self.mDSMEnvelopesManager = DSMEnvelopesManager.init();
        }
    }
    

    func getCachedEnvelopeIds() -> [String]?
    {
        return self.mDSMEnvelopesManager?.cachedEnvelopeIds();
    }


    func syncEnvelopes() -> Void
    {
        self.mDSMEnvelopesManager?.syncEnvelopes();
    }
    
    func presentComposeEnvelopeViewController(_ presentingController:UIViewController, _ signingMode:DSMSigningMode) -> Void
    {
        // you can use method hasCachedDraftEnvelope() to determine whether you'd like to resume a draft or not
        self.mDSMEnvelopesManager?.presentComposeEnvelopeController(withPresenting:presentingController, signingMode:signingMode, resumeWithDraft: true, animated: true, completion: nil)
    }
    
}
